src/main/java/at/hannibal2/skyhanni/features/inventory/DojoRankDisplay.kt

```diff
@@ -0,0 +1,86 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.indexOfFirst
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class DojoRankDisplay {
+
+    private val config get() = SkyHanniMod.feature.inventory
+    private var display = emptyList<String>()
+    private val patternGroup = RepoPattern.group("inventory.dojo.rankdisplay")
+    private val testNamePattern by patternGroup.pattern("name", "(?<color>§\\w)Test of (?<name>.*)")
+    private val testRankPattern by patternGroup.pattern("rank", "(?:§\\w)+Your Rank: (?<rank>§\\w.) §8\\((?<score>\\d+)\\)")
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        config.dojoRankDisplayPosition.renderStrings(display, posLabel = "Dojo Rank Display")
+    }
+
+    private fun drawDisplay(items: Map<Int, ItemStack>) = buildList {
+        var totalScore = 0
+        for ((_, stack) in items) {
+            val name = stack.displayName ?: continue
+            testNamePattern.matchMatcher(name) {
+                val testColor = group("color")
+                val testName = group("name")
+                for (line in stack.getLore()) {
+                    testRankPattern.matchMatcher(line) {
+                        val rank = group("rank")
+                        val score = group("score").toInt()
+                        val color = if (score in 0 .. 99) "§c" else "§a"
+                        totalScore += group("score").toInt()
+                        add("$testColor$testName§f: $rank §7($color${score.addSeparators()}§7)")
+                    }
+                }
+            }
+        }
+
+        // TODO: use repo
```

Comment:
Do we want to create the repo for this right now or later? im happy with both. just make sure its actually planned and not forgetting it.